include/cute/atom/mma_atom.hpp

```diff
@@ -599,6 +599,46 @@ make_tiled_mma(MMA_Op       const&,
   return make_tiled_mma(MMA_Atom<MMA_Op>{}, thr_layout, permutations);
 }
 
+// This helper fn adopts the approach described in
+// media/docs/cute/0t_mma_atom.md#tiledmmas to construct a scatter
+// permutation which ensures all sub-groups operate on contiguous
+// chunks of the MMA. The docs describe how the Layout
+// (i.e. SubgroupLayout) implies a repetition of the atom across additional
+// hardware. Permutations, in the simplest form, imply additional iterations
+// to cover a larger tile (i.e. CTATileShape) than the hardware can handle
+// at once. By specifying additional modes and strides, we can achieve 
+// arbitrary TiledMMA layouts.
+template <typename MMA_Atom, typename CTATileShape, typename SubgroupLayout>
+struct ContigBlockMMAHelper{
+private:
+  using AtomShape = typename MMA_Atom::Shape_MNK;
+
+  // Represents the canonical MMA block which would be handled
+  // by these subgroups without permutation info...
+  static constexpr auto DefaultBlock =
+      blocked_product(Layout<AtomShape>{}, SubgroupLayout{});
+
+  // Discard strides and internal modes, e.g. (8,8):(1,8) -> 64
+  static constexpr auto DefaultTiler =
+      make_shape(size<0>(DefaultBlock), size<1>(DefaultBlock),
+                size<2>(DefaultBlock));
+
+  // Construct the default tiled MMA, to extract the iteration count per dim below
+  static constexpr auto DefaultTiling = logical_divide(Layout<CTATileShape>{}, DefaultTiler);
+
+  static constexpr auto permutation =
+      transform(AtomShape{}, SubgroupLayout{}.shape(), DefaultTiling.shape(),
+                [](auto a, auto sg, auto dt) {
+                  constexpr auto iters = get<1>(dt);
+                  auto tiler_shape = make_shape(a, sg, iters);                     // atom, hardware, iteration
```

Comment:
I dont like the term hardware here. Atom is what is done by the hardware. How about tiling or tile?

Reply:
The issue for me is that both the 2nd and 3rd modes are kinda 'tiling'. The distinction is between 'tiling by involving more hardware' and 'tiling by making each bit of hardware do more iterations'.

We could say something like `// atom, sub-group, iteration` but that ties us to the idea that an atom is always done by a whole sub-group. So far that's only true for PVC as far as I know.

Reply:
In my mind a tile is a chink of work that is worked on concurrently. If chunks are consumed one after another that is iterating over tiles. I agree subgroup here is specific to PVC, but on the other hand hardware is even more wrong term - atom is the only thing that is hardware specific. That is why I suggested tile. 

Or if you want to be more more verbose you could say `tile_over_`[`hw`/`sgs`/`threads`] .

Reply:
Since the comment is meant to serve as a mnemonic for the reader/team, I'd prefer not to be more verbose.

I think this comment is concise, accurate and clear because:
 - since it describes a `tiler_shape` it should be clear to the reader that this is describing a tiling
 - this second mode *always* describes iteration over additional hardware (not just on PVC)
 - since it's preceded by 'atom', IMO it is clear that 'hardware' does not mean 'the hardware involved in a single atom'.

Reply:
For me having hardware and atom here makes me thing there is something weird going on here. I can not rely on the comments and I need to reverse engineer it to understand what it is. In other words I am still strongly opposed to the term.